docs/dg/dev/architecture/coding-conventions.md

```diff
@@ -0,0 +1,2165 @@
+
+```
+[Organization]
+└── Client
+    └── [Module]
+        ├── [CustomDirectory]
+        │   ├── [ModelName]Interface.php
+        │   └── [ModelName].php        
+        ├── Dependency
+        │   ├── Client
+        │   │   ├── [Module]To[Module]ClientBridge.php
+        │   │   └── [Module]To[Module]ClientInterface.php
+        │   ├── Service
+        │   │   ├── [Module]To[Module]ServiceBridge.php
+        │   │   └── [Module]To[Module]ServiceInterface.php           
+        │   └── Plugin
+        │       └── [PluginInterfaceName]PluginInterface.php            
+        ├── Plugin
+        │   └── [ConsumerModule]
+        │       └── [PluginName]Plugin.php
+        ├── Zed
+        │   ├── [Module]Stub.php
+        │   └── [Module]StubInterface.php
+        ├── [CustomDirectory]
+        ├── [Module]ClientInterface.php
+        ├── [Module]Client.php
+        ├── [Module]Config.php
+        ├── [Module]DependencyProvider.php
+        └── [Module]Factory.php
+```
+
+Used components
+- [Bridge](#bridge)
+- [Client facade](#client-facade)
+- [Config](#module-configurations)
+- [Dependency Provider](#dependency-provider)
+- [Factory](#factory)
+- [Mapper / Expander / Hydrator](#mapper--expander--hydrator)
+- [Model](#model)
+- [Plugin](#plugin), [Plugin Interface](#plugin-interface)
+- [Zed Stub](#zed-stub)
+
+## Service
+
+The `Service` application layer encapsulates a multipurpose library that can be utilized across various application layers, such as `Yves` or `Zed` application layers.
+A `Service` primarily consists of reusable lightweight stateless business logic components.
+Due to its deployment across all applications, a `Service` is constrained to accessing data providers that are available universally (for instance, the backend database is not accessible from storefront applications by default).
+
+```
+[Organization]
+└── Service
+    └── [Module]
+        ├── [CustomDirectory]
+        │   ├── [ModelName]Interface.php
+        │   └── [ModelName].php        
+        ├── [Module]Config.php            
+        ├── [Module]DependencyProvider.php
+        ├── [Module]ServiceFactory.php
+        ├── [Module]ServiceInterface.php
+        └── [Module]Service.php
+```
+
+Used components
+- [Config](#module-configurations)
+- [Dependency Provider](#dependency-provider)
+- [Factory](#factory)
+- [Mapper / Expander / Hydrator](#mapper--expander--hydrator)
+- [Model](#model)
+- [Service facade](#service-facade)
+
+## Shared
+
+The `Shared` library contains code that is designed to be utilized across different application layers and modules.
+The `Shared` library is intended to facilitate the sharing of code among various application layers and modules.
+To ensure compatibility and versatility across different application architecture setups, any content within the `Shared` library must be free of application layer-specific elements. Therefore, the use of [Factories](#factory) is not permitted within the `Shared` library.
```

Comment:
mb we can also mention that shared config can be accessed from Application Layer specific configs